ocaml/xapi/storage_smapiv3_migrate.ml

```diff
@@ -12,30 +12,322 @@
  * GNU Lesser General Public License for more details.
  *)
 
-module D = Debug.Make (struct let name = "storage_smapiv1_migrate" end)
+module D = Debug.Make (struct let name = __MODULE__ end)
 
 module Unixext = Xapi_stdext_unix.Unixext
 module State = Storage_migrate_helper.State
 module SXM = Storage_migrate_helper.SXM
+open Storage_interface
+open Storage_task
+open Xmlrpc_client
+open Storage_migrate_helper
 
 module type SMAPIv2_MIRROR = Storage_interface.MIRROR
 
+let s_of_sr = Storage_interface.Sr.string_of
+
+let s_of_vdi = Storage_interface.Vdi.string_of
+
+let s_of_vm = Storage_interface.Vm.string_of
+
+let export_nbd_proxy ~remote_url ~mirror_vm ~sr ~vdi ~dp ~verify_dest =
+  D.debug "%s spawning exporting nbd proxy" __FUNCTION__ ;
+  let path =
+    Printf.sprintf "/var/run/nbdproxy/export/%s" (Vm.string_of mirror_vm)
+  in
+  let proxy_srv = Fecomms.open_unix_domain_sock_server path in
+  try
+    let uri =
+      Printf.sprintf "/services/SM/nbdproxy/import/%s/%s/%s/%s"
```

Comment:
Should this use the URI module?

Reply:
I could, but it is a bit awkward with the `Http` module which itself is a wrapper around `Uri`